README.md

```diff
@@ -30,7 +30,7 @@ If your environment was restarted or has been inactive over some time, you might
 
 ## Testing your Project
 
-To run the [integration tests](/src/icp_hello_world_rust_backend/tests/integration_tests.rs#L18) for your backend canister, first run `./scripts/build.sh` to build the canister, and then `cargo test --test integration_tests`. 
+To run the [integration tests](/src/icp_hello_world_rust_backend/tests/integration_tests.rs#L18) for your backend canister, first run `dfx deploy` to build the canister Wasm, and then `cargo test --test integration_tests`. 
```

Comment:
What if I don't want to launch dfx? Is there no dfx build command that has no other effect than the necessary one?

Reply:
Also,
>custom build script did not include the candid metadata field
can we not fix this in the build script and add the metadata there?

Reply:
We could add it in the metadata manually, but what's the point? I think it'll add unnecessary complexity. If the default works, I think we can stick with it

Reply:
`dfx build` alone would not suffice to build the wasm, one would need `dfx canister create --all
dfx build`. To keep the instructions for new users to a minimum, I'd suggest to keep `dfx deploy`.

Reply:
The default works? You mean using dfx? 
Asked differently, can we not make it work without invoking dfx? I'd think that cargo build should somehow be able to achieve the goal, given the right configuration. Something like
```
cargo build --target=...wasm... --additional-args-for-metadata-inclusion
cargo test --test integration tests
```
dfx does it somehow, so it is possible.

The reasons I prefer this are:
- it looks like dfx and pocket-ic are somehow dependent on each other, which is not an impression I want to make, especially for newcomers who don't know what means what. 
- you need to launch a background process (dfx) that is completely unnecessary for running the integration test
- the command dfx deploy implies and does something completely different than building a canister, it is merely a sideeffect that the canister wasm ends up on the disk in a place our integration test can then use. in fact, if you repeat the command, it might fail depending on the state of the dfx process, and you'd need to pass --clean. but really you just wanted to compile your canister to wasm.

Reply:
(Use the default was w.r.t. to the candid metadata and the build process)

`dfx deploy` does a lot of things in one command: potentially shrinking the wasm with `ic-wasm`, gzipping it, adding metadata and more.
To replicate the same process with cargo is tedious and if not done right can easily lead to different wasms that come out at the end, which in turn can lead to testing a different wasm that is not gonna be the same that will be produced by `dfx`, which will be the wasm updated to mainnet. I would therefore not encourage users to compile the wasm differently (we don't know how their dfx.json will end up looking and what steps are necessary), so `dfx deploy` is the only safe way I see.

Reply:
good point, it is desirable to have the same artifact. what about `dfx build`? then at least two of my concerns can be addressed. 
I still don't like the dependency, but we live with it for now (personally I would just replicate the steps that dfx does in the script, so people can see what is going on. but it's your call. you say the build depends on the dfx.json, what for example?).

Reply:
ah you answered that above

Reply:
but I definitely prefer the two lines over dfx deploy.

Reply:
Alright, then let me update the readme and `.except()` to recommend these two lines 
```bash
dfx canister create --all
dfx build
```